skimage/feature/corner.py

```diff
@@ -154,18 +159,29 @@ def hessian_matrix(image, sigma=1, mode='constant', cval=0):
     gaussian_filtered = ndi.gaussian_filter(image, sigma=sigma,
                                             mode=mode, cval=cval)
 
+    if order is None:
+        if image.ndim == 2:
+            # The legacy 2D code followed (x, y) convention, so we swap the axis
+            # order to maintain compatibility with old code
+            warn('deprecation warning: the default order of the hessian matrix values '
+            'will be "row-column" instead of "xy" starting in skimage version 0.15'
+             'Use order="rc" or order="xy" to set this explicitly')
+            order = 'rc'
```

Comment:
Sorry, i meant:

if order is None:
    if image.ndim == 2:
        # The legacy 2D code followed (x, y) convention, so we swap the axis
        # order to maintain compatibility with old code
        warn('deprecation warning: the default order of the hessian matrix values '
        'will be "row-column" instead of "xy" starting in skimage version 0.15'
         'Use order="rc" or order="xy" to set this explicitly')
        order = 'rc'
    else:
        order = 'xy'



if order == 'rc':
    axes = reversed(axes)